addons/sale_stock/models/sale_order.py

```diff
@@ -157,7 +157,7 @@ def action_cancel(self):
             if sale_order.state == 'sale' and sale_order.order_line:
                 sale_order_lines_quantities = {order_line: (order_line.product_uom_qty, 0) for order_line in sale_order.order_line}
                 documents = self.env['stock.picking']._log_activity_get_documents(sale_order_lines_quantities, 'move_ids', 'UP')
-        self.mapped('picking_ids').action_cancel()
+        self.mapped('picking_ids').filtered(lambda p: p.state != 'done').action_cancel()
```

Comment:
This change from `sale_stock` sneaked in accidentally?

Reply:
During testing, I set my warehouse in 2 steps delivery. Marked as done my first transfer. At this time I could not cancel the sales order. We made this fix better in V14+ (see 4c36a35bf14f) but I though this was acceptable to put in v13.0

Reply:
I see... I suppose it's fine then. This just looked out of place 😛

Reply:
Hi, cancel a picking done is wrong for me, is inconsistent(sale order `canceled `and stock `validated`), please consider keep as V12(at least on stable version)

V12

![imagen](https://user-images.githubusercontent.com/66845741/131444351-3c45ddf3-1413-46e9-b9bf-e87473a29fbe.png)